### DISCUSSION TITLE
Difficulty Persisting User Sessions and Accessing Data

### DISCUSSION THREAD

**Discussion by Valgysuma** [supabase/supabase#30646] [Questions]
I am writing to report issues encountered during the development of my project using the Supabase platform. My application, Waakuva, is a motorcycle-based urban transportation service that relies on Supabase for user authentication and data management.

Currently, I am facing the following challenges:

Session Persistence: Despite implementing the recommended methods from the Supabase documentation, user sessions are not being persisted correctly. This forces users to log in every time they open the app.
Data Access: Although user data is successfully registered in the database, I am unable to retrieve it consistently. Queries to the user table return empty or inconsistent results, even when records exist.
I have followed the best practices outlined in your documentation and thoroughly reviewed my implementation. Believing the issue might stem from Supabase configurations, I kindly request your assistance in identifying and resolving these issues.

My project depends on reliable session persistence and data access to provide a seamless user experience. I would appreciate your support in resolving these challenges promptly.

**Comment by GaryAustin1** [comment]
You will likely need to provide more info on your issues and some possible code sample.

Supabase does persist sessions as long as the user does not signout and their local storage or cookies (if using SSR as you don't even say your framework) is not removed.  

You also mention not getting data which sounds like RLS issues, usually not meeting policy.  An empty array on a select is the expected response if you are not signed in and have RLS requiring an authenticated user or no RLS policy set/met.

There are no general bugs with this process.
